app/models/mixins/event_mixin.rb

```diff
@@ -36,10 +43,43 @@ def events_table_name(assoc)
     events_assoc_class(assoc).table_name
   end
 
+  def event_where_clause(assoc = :ems_events)
+    case assoc.to_sym
+    when :ems_events, :event_streams
+      event_where_clause_ems_events
+    when :miq_events, :policy_events
+      event_where_clause_miq_events
+    end
+  end
+
+  def event_where_clause_ems_events
+    EmsEvent.where(belongs_to_column_id => id)
+  end
+
+  def event_where_clause_miq_events
+    MiqEvent.where(belongs_to_column_id => id)
+  end
+
   private
 
   def find_one_event(assoc, order)
     ewc = event_where_clause(assoc)
     events_assoc_class(assoc).where(ewc).order(order).first unless ewc.blank?
   end
+
+  def belongs_to_column_id
```

Comment:
Should this be memoized and/or done at the class level?  Unless I'm misreading, nothing in this method uses any instance information.

Reply:
yes, it can be